src/WebJobs.Extensions.DurableTask.Analyzers/SyntaxNodeUtils.cs

```diff
@@ -29,16 +29,34 @@ public static DurableVersion GetDurableVersion(SemanticModel semanticModel)
 
         public static bool TryGetSemanticModelForSyntaxTree(SemanticModel model, SyntaxNode node, out SemanticModel newModel)
         {
-            if (model?.SyntaxTree == null || node?.SyntaxTree == null)
+            try
+            {
+                if (model?.SyntaxTree == null || node?.SyntaxTree == null)
+                {
+                    newModel = null;
+                    return false;
+                }
+
+                var compilation = model.Compilation;
+                if (!compilation.ContainsSyntaxTree(node.SyntaxTree))
+                {
+                    var newComplilation = model.Compilation.AddSyntaxTrees(node.SyntaxTree);
+                    newModel = newComplilation.GetSemanticModel(node.SyntaxTree);
+                }
+                else
+                {
+                    newModel = model.SyntaxTree == node.SyntaxTree
+                    ? model
+                    : model.Compilation.GetSemanticModel(node.SyntaxTree);
+                }
+
+                return newModel != null;
+            }
+            catch( ArgumentException e) when (e.Message == "Inconsistent language versions")
```

Comment:
I couldn't find a good way to check this so I had to catch this exception.

Reply:
When do we expect to hit this exception? a comment describing when we expect to hit this control flow would be super helpful.